### ISSUE TITLE
Prosody configuration does not set trusted proxies directive

### ISSUE THREAD

**Issue by millaguie** [jitsi/docker-jitsi-meet#1042]
According to [Prosody documentation](https://prosody.im/doc/http#trusted_proxieses) trusted reverse proxies has to be set in order to be able to know the real IP address of a client. This is preventing the log_auth module to show the real IP address of the client as it always shows the IP address of the web server (who is acting as a proxy).

There is no easy way to solve this issue, as it depends on the IP address of another container, a workaround is to set a depends_on tag on the prosody container forcing it to start after the web service, and change the configuration script to resolve the web server and set that IP (or IPs) in trusted_proxies. Another option is patch the prosody http_module to accept CIDRs (currently it only accept a list of IP address) and set the docker CIDR as trusted.

Has somebody else face this problem? 

Thanks!

**Comment by saghul** [comment]
Good point. I wonder if setting the trusted proxy to `"*"` would work.

**Comment by saghul** [comment]
In this setup clients never connect directly to prosody, it always goes through the web container.

Does your concern still apply in this scenario @bgrozev ?

**Comment by bgrozev** [comment]
@saghul yes, a client could just include the header in the original request, and prosody will trust it.

@millaguie I just remembered we [added support for CIDR to prosody](https://hg.prosody.im/trunk/rev/dff1aebd0f2b) for just this reason. I don't know if it's made it to 0.11 or the version used here though.

**Comment by saghul** [comment]
> @saghul yes, a client could just include the header in the original request, and prosody will trust it.

We'd set that header ourselves when proxying inwards, of course.

**Comment by bgrozev** [comment]
> We'd set that header ourselves when proxying inwards, of course.

Most proxies are configured to append to it

**Comment by saghul** [comment]
I'm a bit confused :-) This setup has Nginx in front of Prosody. Prosody is not exposed directly, ever. If we set this header in Nginx and configure Prosody to trust anything, do you still have concerns?

**Comment by saghul** [comment]
Gotcha.

**Comment by millaguie** [comment]
Hi,

Thanks @bgrozev! 

> @millaguie I just remembered we [added support for CIDR to prosody](https://hg.prosody.im/trunk/rev/dff1aebd0f2b) for just this reason. I don't know if it's made it to 0.11 or the version used here though.

Yes, I think this is the proper way, but that code is not in 0.11.x only in trunk :(

**Comment by saghul** [comment]
> Yes, I think this is the proper way, but that code is not in 0.11.x only in trunk :(

Ah, that's unfortunate. Once it lands in a release, holler and we'll update and make it configurable.
